### PR TITLE
Core: Exclude some keyboard keys to prevent revalidate the field

### PR DESCRIPTION
**Pull request by Arkni**
Avoid revalidate the field when pressing one of the following keys in `onkeyup` method:
```
	Shift           => 16
	Alt             => 18
	Caps lock       => 20
	End             => 35
	Home            => 36
	Left arrow      => 37
	Up arrow        => 38
	Right arrow     => 39
	Down arrow      => 40
	Insert          => 45
	Num lock        => 144
	AltGr key       => 225
```

### CODE REVIEW

**Comment by staabm** [comment]
Whats the reasoning behind this change?

**Comment by Arkni** [comment]
These keys are used just to navigate, change character's case ...

This change is just a small enhancement to gain in performance and avoiding calling the
`this.element( element )` method every time these keys are pressed.

**Comment by staabm** [comment]
Sounds good to me. To land it requires a unit test which tests at least one of those "special-characters".

@jzaefferer are you ok with this change?

File: src/core.js [hunk 1]
```diff
@@ -259,7 +259,25 @@ $.extend( $.validator, {
 			}
 		},
 		onkeyup: function( element, event ) {
-			if ( event.which === 9 && this.elementValue( element ) === "" ) {
+			// Avoid revalidate the field when pressing one of the following keys
+			// Shift       => 16
```
Comment (staabm) [inline src/core.js]: What about `ctrl`

Comment (Arkni) [inline src/core.js]: The  `ctrl` key trigger the `keyup` event when used with another key as combination ( `ctrl + c`, `ctrl + v`, ...)
Do we have to check for it also ?

**Comment by Arkni** [comment]
@staabm 
which file I have to edit in order to add a test suite for this change ?

**Comment by jzaefferer** [comment]
This looks reasonable, should help especially with the remote method.

Since Ctrl by itself doesn't change the input value, it seems good to exclude (add to the list) as well.

As for unit tests: There are two tests in `test/test.js` that set the `onkeyup` option and trigger `keyup` events. Here you'd also need to set the `keyCode` property, like this:

```js
var e = jQuery.Event( "keyup", { keyCode: 16 } );
```

**Comment by Arkni** [comment]
@jzaefferer 

Done.

File: src/core.js (hunk 1 above)
Comment (jzaefferer) [inline src/core.js]: This is resolved, Ctrl was added.

**Comment by jzaefferer** [comment]
Nice! Squashed and merged, with a small update of the commit message.

**Pull request closed without merging**